### DISCUSSION TITLE
docker_remote.image.load(file) local or remote path?

### DISCUSSION THREAD

**Discussion by s-light** [gabrieldemarmiesse/python-on-whales#622] [Q&A] [answered]
hello community,

i am in the process to convert my (bad) bash-script for manual deploys to a python version.
and with this i am using python-on-whales.

i have two docker clients - my local on and the remote one.
the remote is initialized with
```
self.docker_remote = DockerClient(host=self.server_address)
```
( i bundled my script in a simple class..)

and with this i tried to use the [image.load](https://gabrieldemarmiesse.github.io/python-on-whales/sub-commands/image/#python_on_whales.components.image.cli_wrapper.ImageCLI.load)
tool. but it fails with 

> open /var/www/web1/docker-compose-folder/temp/currentDeploy_WIP.tar: no such file or directory

if i check on the server the file is there:
```
:~/docker-compose-mis-prod/temp$ ll currentDeploy_WIP.tar 
-rw------- 1 web1 web1 6,2G 19. Aug 13:57 currentDeploy_WIP.tar
```

so is the path given here a local path?
or have i overlooked something else?

sunny greetings
stefan

**Comment by gabrieldemarmiesse** [comment, marked as answer]
I believe that docker load always use a local path and never a remote path. In general, you can never read files on the remote host (docker cp, docker build all use local paths). Containers can have access to the remote files of course with volumes, but that's not what you want.

I don't have any solution to read a remote path. Thats a docker issue, not a python-on-whales issue. You'd have the same problem when using "docker --host=... load" from the command line.

**Reply by s-light** [reply]
thanks for your reply Gabriel!

yeah seems that was one of the things i did miss entirely in the first place...

when i use a local path it is kind of working - 
it stops with an 
`Post "http://the.domain.de:2375/v1.45/images/load?quiet=1": dial tcp [xx00:61xx:101:2x2::1]:2375: i/o timeout`
i have no idea what on my setup is wrong here - 
and it is a docker client error - so nothing from your library...

as the image is currently very big (6GB - will be optimized later)  i think timeouts could be very normal..

two options come to my mind:
1. system tools
 i just do  subprocess call with the system rsync for upload and then ssh commands for the docker client on the server..
its a easy thing...

2. feels *nicer* but for now to complex to test:
 use the [example from the documentation with save and load](https://gabrieldemarmiesse.github.io/python-on-whales/sub-commands/image/#python_on_whales.components.image.cli_wrapper.ImageCLI.save--example) 
 this uses a bytes_iterator in between..
 and maybe i could get some sort of progressbar with [tqdm](https://tqdm.github.io/) for this.. 
 i have no idea - and could be hard as i do not know the size upfront..
